`koch web`: parallelization does not work (see current travis failures)

The command `koch web` tries to build the documentation in parallel using all available processors. But this always fails, and then the work is repeated serially. The parallel build of the documentation always fails, because today building the documentation involves running the `runnableExamples`. But this does not work in parallel, because the multiple running compilers will eventually clobber each others nimcache.
We therefore regularly see  messages like `undefined reference to <whatever>` eg. in the build logs of travis, but also if we run `koch web` locally and redirect the output to a file. We also see the misterious message `external program failed, retrying serial work queue for logs!` in the output of `koch web` which marks the switch from parallel to serial work.
Until someone comes up with a solution to prevent the race condition on the nimcache, when multiple compilations run in parallel, it is probably advisable to disable the parallelization in [nimweb.nim](https://github.com/nim-lang/Nim/blob/devel/tools/nimweb.nim#L320) by changing line 320 from 
```nim
 mexec(commands, c.numProcessors)
```
to
```nim
 mexec(commands, 1)
```

--- a/compiler/sem.nim
+++ b/compiler/sem.nim
@@ -620,14 +620,11 @@ proc testExamples(c: PContext) =
     quit "[Examples] failed"
   else:
     removeFile(outp)
-    when defined(windows):
-      removeFile(outp.changeFileExt"" & ".exe")
-    else:
-      removeFile(outp.changeFileExt"")
-      try:
-        removeDir(nimcache)
-      except:
-        discard
+    removeFile(outp.changeFileExt(ExeExt))
+    try:
+      removeDir(nimcache)
+    except OSError:
+      discard
 
 proc myClose(graph: ModuleGraph; context: PPassContext, n: PNode): PNode =
   var c = PContext(context)

--- a/compiler/sem.nim
+++ b/compiler/sem.nim
@@ -610,14 +610,24 @@ proc myProcess(context: PPassContext, n: PNode): PNode =
 proc testExamples(c: PContext) =
   let inp = toFullPath(c.config, c.module.info)
   let outp = inp.changeFileExt"" & "_examples.nim"
+  let nimcache = outp.changeFileExt"" & "_nimcache"
   renderModule(c.runnableExamples, inp, outp)
   let backend = if isDefined(c.config, "js"): "js"
                 elif isDefined(c.config, "cpp"): "cpp"
                 elif isDefined(c.config, "objc"): "objc"
                 else: "c"
-  if os.execShellCmd(os.getAppFilename() & " " & backend & " -r " & outp) != 0:
+  if os.execShellCmd(os.getAppFilename() & " " & backend & " --nimcache:" & nimcache & " -r " & outp) != 0:
     quit "[Examples] failed"
-  removeFile(outp)
+  else:
+    removeFile(outp)
+    when defined(windows):
+      removeFile(outp.changeFileExt"" & ".exe")
+    else:
+      removeFile(outp.changeFileExt"")
+      try:
+        removeDir(nimcache)
+      except:
+        discard
 
 proc myClose(graph: ModuleGraph; context: PPassContext, n: PNode): PNode =
   var c = PContext(context)

